citrination_client/models/client.py

```diff
@@ -86,16 +87,27 @@ def predict(self, data_view_id, candidates, method="scalar", use_prior=True):
                 "Prediction failed: UID={}, result={}".format(uid, result["status"])
             )
 
-    def retrain(self, dataview_id):
+    def retrain(self, data_view_id, is_async=True, timeout=TRAIN_TIMEOUT):
         """
         Start a model retraining
-        :param dataview_id: The ID of the views
+        :param data_view_id: The ID of the views
+        :param is_async: Whether or not to make this call asynchronously
+        :type is_async: bool
+        :param timeout: Number of seconds to wait, if not async
+        :type timeout: int
         :return:
         """
-        url = 'data_views/{}/retrain'.format(dataview_id)
+        url = 'data_views/{}/retrain'.format(data_view_id)
         response = self._post_json(url, data={})
         if response.status_code != requests.codes.ok:
             raise RuntimeError('Retrain requested ' + str(response.status_code) + ' response: ' + str(response.message))
+        
+        if not is_async:
+            self._wait_for(
+                "Predict services ready",
+                lambda: self.get_data_view_service_status(data_view_id).predict.ready,
```

Comment:
This isn't necessarily correct. Predict status will be `ready=True` if you have a view that you trained previously even if you add new data and call `retrain` again. 

I dealt with this in the stress tests by first waiting for `progress<100` _then_ waiting for `progress = 100`.